agimus_controller/agimus_controller/ocp_base.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+from abc import ABC, abstractmethod
+from agimus_controller.trajectory import ReferenceTrajectory
+from agimus_controller.trajectory import ReferenceTrajectoryPoint
+from agimus_controller.mpc_output import MPCOutput
+
+
+class OCPBase(ABC):
+    def __init__(self) -> None:
+        self.mpc_output_data = MPCOutput()
+        self.mpc_debug_data = MPCOutput()
+
+    @abstractmethod
+    def update_horizon_from_reference(
+        self, reference_trajectory: ReferenceTrajectory
+    ) -> bool:
+        return False
+
+    @abstractmethod
+    def get_debug_infos(self):
+        return
+
+    @abstractmethod
+    def get_x_from_trajectory_point(
+        self, trajectory_point: ReferenceTrajectoryPoint
+    ) -> np.ndarray:
+        pass
+
+    @abstractmethod
+    def get_u_from_trajectory_point(
+        self, trajectory_point: ReferenceTrajectoryPoint
+    ) -> np.ndarray:
+        pass
+
+    @abstractmethod
+    def get_horizon_size() -> int:
+        pass
+
+    # @abstractmethod ?
```

Comment:
I think there can be a default implementation that just passes everything to Crocoddyl classes and then solves the problem. If someone needs custom version of this then they can just overwrite it